collector/src/main/java/io/prometheus/jmx/JmxScraper.java

```diff
@@ -89,7 +90,16 @@ public void doScrape() throws Exception {
         MBeanServerConnection beanConn;
         JMXConnector jmxc = null;
         if (jmxUrl.isEmpty()) {
-          beanConn = ManagementFactory.getPlatformMBeanServer();
+          try {
```

Comment:
I thought you might say that... no problem, will revert. At best this whole thing is a work-around anyway.